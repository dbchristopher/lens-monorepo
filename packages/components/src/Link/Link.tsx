import React from 'react';
import styled from 'styled-components';

export interface LinkProps {
  id: 'string';
}

/** @component */
export const Link = styled.a``;
