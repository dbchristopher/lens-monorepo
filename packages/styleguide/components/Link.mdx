---
name: Link
---

import {Playground} from 'docz';
import {Link} from '@project/components';

### Default Link

The `<Link />` component renders an `<a>` tag that accepts and `href` property. You can also supply an optional `id` property if you want to give your anchor an id.

<Playground>
  <Link href="https://looker.com">👋 I am a link!</Link>
  <Link href="https://looker.com" id="thumbs-up">
    👍 I am a link with an id
  </Link>
</Playground>
