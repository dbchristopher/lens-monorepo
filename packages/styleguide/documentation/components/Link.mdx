---
name: Link
menu: Components
---

import { Playground } from 'docz';
import { List, ListItem, Link } from 'looker-lens-components';

# Link

The `<Link />` component renders an `<a>` tag that accepts and `href` property. You can also supply an optional `id` property if you want to give your anchor an id.

<Playground>
  <List>
    <ListItem>
      <Link href="https://looker.com">👋 I am a link!</Link>
    </ListItem>
    <ListItem>
      <Link href="https://looker.com" id="thumbs-up">
        👍 I am a link with an id
      </Link>
    </ListItem>
  </List>
</Playground>
