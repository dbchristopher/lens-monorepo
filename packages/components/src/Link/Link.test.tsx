import 'jest-styled-components'
import * as React from 'react'
import { createWithTheme } from '../../../test/utils/create_with_theme'
import { Link } from './Link'

test('A default Link', () => {
  const component = createWithTheme(<Link href="https://looker.com">ðŸ¥‘</Link>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('A external Link', () => {
  const component = createWithTheme(<Link href="https://looker.com">ðŸ¥‘</Link>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('A Link with an id', () => {
  const component = createWithTheme(
    <Link href="https://looker.com" id="link-id">
      ðŸ¥‘
    </Link>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
