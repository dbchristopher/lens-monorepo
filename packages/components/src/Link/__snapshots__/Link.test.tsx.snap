// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Link with an id 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  color: #0087e1;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
  className="c0"
  href="https://looker.com"
  id="link-id"
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  ðŸ¥‘
</a>
`;

exports[`A default Link 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  color: #0087e1;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
  className="c0"
  href="https://looker.com"
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  ðŸ¥‘
</a>
`;

exports[`A external Link 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  color: #0087e1;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
  className="c0"
  href="https://looker.com"
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  ðŸ¥‘
</a>
`;
